src/dataLoader/tagsDataLoader.ts

```diff
@@ -1,55 +1,110 @@
+import config from '../config';
 import { IContext } from '../server/context';
-import DataLoader from 'dataloader';
+import { reorderResultByKey } from './utils';
 import { SavedItemDataService, TagDataService } from '../dataService';
 import { Tag } from '../types';
-import { reorderResultByKey } from './utils';
 import { TagModel } from '../models';
+import DataLoader from 'dataloader';
 
 /**
  * Create dataloaders to cache and batch requests for Tags made
  * in a single tick of the application.
- * There are two loaders for Tags which are differentiated by
- * keys: one accesses the Tag by ID, and one by name. Each loader
- * fills the cache of the other when loading from either key (since they
- * refer to the same object, just via alternative keys).
+ * There are three loaders for Tags.
+ * 2 loaders differ by keys: one accesses Tag by ID, & the other by name.
+ * The third loader accesses all Tags associated with an Item via the Item's ID.
+ * The first 2 loaders fill the cache of the other when loading from either key
+ * (since they refer to the same object, just via alternative keys).
  * That way resolving the same object by alternative key does not result
  * in duplicate fetches.
+ * The third loader does not cache, as its primary use is query batching
+ * in order to handle a n+1 problem when wanted all Tags for a list of Items.
  * @param context IContext object with database connection. Should
  * be freshly created for every GraphQL request.
  */
 export function createTagDataLoaders(
   context: IContext
-): Pick<IContext['dataLoaders'], 'tagsById' | 'tagsByName'> {
+): Pick<IContext['dataLoaders'], 'tagsById' | 'tagsByName' | 'tagsByItemId'> {
   const byIdLoader = new DataLoader(async (ids: string[]) => {
-    const tags = await batchGetTagsByIds(context, ids);
+    const savedItemdService = new SavedItemDataService(context);
+    const tags = await batchGetTagsByIds(
+      new TagDataService(context, savedItemdService),
+      ids
+    );
     tags.forEach((tag) => byNameLoader.prime(tag.name, tag));
     return tags;
   });
   const byNameLoader = new DataLoader(async (names: string[]) => {
-    const tags = await batchGetTagsByNames(context, names);
+    const savedItemdService = new SavedItemDataService(context);
+    const tags = await batchGetTagsByNames(
+      new TagDataService(context, savedItemdService),
+      names
+    );
     tags.forEach((tag) => byIdLoader.prime(tag.id, tag));
     return tags;
   });
-  return { tagsById: byIdLoader, tagsByName: byNameLoader };
+  const byItemIdLoader = new DataLoader(
+    async (itemIds: string[]) => {
+      const savedItemdService = new SavedItemDataService(context);
+      const tags = await batchGetTagsByItemIds(
+        new TagDataService(context, savedItemdService),
+        itemIds
+      );
+      return tags;
+    },
+    {
+      maxBatchSize: config.dataloaderDefaults.maxBatchSize,
```

Comment:
do you need `cache=False` here for it to not cache?

Reply:
the cache push / check is in a diff util function that this batch function doesn't use iirc; will confirm.

Reply:
yeah okay - so for the other tags dataloaders, they call a `tags.forEach((tag) => byIdLoader.prime(tag.id, tag));` in their dataloader function, which then writes to the cache setup by default by dataLoader class. 

I don't do any of that for the byItemId dataloader - so it does look like a cache is created for it by DataLoader class by default, but we just never write to it. Going ahead and also explicitly disabling even making the unused cache to start with too.